src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -54,13 +54,9 @@ public override void MoveFile(string sourceFullPath, string destFullPath)
         {
             // The desired behavior for Move(source, dest) is to not overwrite the destination file
             // if it exists. Since rename(source, dest) will replace the file at 'dest' if it exists,
-            // a check is added before rename is called. Note that this is a race condition. Possible
-            // better solutions would be to use link/unlink or platform specific functions instead of
-            // rename(source, dest). Regardless of the implementation, an IOException should be thrown if 
-            // 'dest' refers to a file that already exists.
-            if (File.Exists(destFullPath))
-                throw new IOException(SR.GetResourceString(SR.IO_AlreadyExists_Name, destFullPath));
-            while (Interop.libc.rename(sourceFullPath, destFullPath) < 0)
+            // link/unlink are used instead. Note that the Unix FileSystemWatcher will treat a Move 
```

Comment:
This is a problem for the FSW...we shouldn't have functional differences for the same IO operation; callers of the FSW will respond differently to Create/Delete events then they will to Move events. Can we look into a different way of implementing this function that won't cause a Delete + Create?